src/core_plugins/kibana/public/dashboard/grid.js

```diff
@@ -34,25 +68,25 @@ app.directive('dashboardGrid', function ($compile, Notifier) {
 
       // debounced layout function is safe to call as much as possible
       const safeLayout = _.debounce(layout, 200);
-
-      $scope.removePanelFromState = (panelIndex) => {
-        _.remove($scope.state.panels, function (panel) {
-          return panel.panelIndex === panelIndex;
-        });
-      };
-
       /**
-       * Removes the panel with the given index from the $scope.state.panels array. Does not
-       * remove the ui element from gridster - that is triggered by a watcher that is
-       * triggered on changes made to $scope.state.panels.
-       * @param panelIndex {number}
+       * Mapping of panelIndex to the angular element in the grid.
        */
-      $scope.getPanelByPanelIndex = (panelIndex) => {
-        return _.find($scope.state.panels, function (panel) {
-          return panel.panelIndex === panelIndex;
-        });
+      const panelElementMapping = {};
+
+      // Tell gridster to remove the panel, and cleanup our metadata
+      function removePanelFromGrid(panelElement, silent) {
+        // remove from grister 'silently' (don't reorganize after)
+        gridster.remove_widget(panelElement, silent);
+      }
+
+      $scope.removePanel = (panelIndex) => {
```

Comment:
Both this scope and the parent having a function called `removePanel` is an example of why using controllers and `bindToController` is a good idea. If these functions were bound to a controller the views could qualify which function they would call. I'm not saying that this should be refactored within this PR, but maybe it's something to consider for the future.